src/Slider/Slider.js

```diff
@@ -0,0 +1,190 @@
+import React, { Component } from 'react';
+import SliderStore from './Slider-store';
+import Labels from './Labels';
+
+import './Slider.css';
+
+class Slider extends Component {
+	constructor(props){
+		super(props);
+    this.state = { current : 0, isHover : false };
+    this.handleLeftClick = this.handleLeftClick.bind(this)
+    this.handleRightClick = this.handleRightClick.bind(this)
+    this.handleMouseEnter = this.handleMouseEnter.bind(this)
+    this.handleMouseLeave = this.handleMouseLeave.bind(this)
+	}
+
+  handleLeftClick(){
+    var cs = this.state.current;
+    cs -= 1;
+    if (cs < 0) {
+      cs = SliderStore.length - 1;
+    }
+    this.setState({current:cs});
+  }
+
+  handleRightClick(){
+     var cs = this.state.current;
+    cs += 1;
+    if (cs >= SliderStore.length) {
+      cs = 0
+    }
+    this.setState({current:cs});
+  }
+
+  handleMouseEnter(){
+    this.setState({
+      isHover: !this.state.isHover
+    })
+  }
+
+  handleMouseLeave(){
+    this.setState({
+      isHover: !this.state.isHover
+    })
+  }
+
+  certainSlide = number => {
+    console.log(number)
+    this.setState({current:number});
+  }
+
+  clearIntervals () {
+      this.intervals.map(clearInterval);
+      this.intervals = [];
+  }
+
+  componentWillMount() {
```

Comment:
the lifecycle hook will be deprecated in new version of react please use componentDidMount instead.